plugins/defaultplugins/aclplugin/vppcalls/interfaces_vppcalls.go

```diff
@@ -14,144 +14,221 @@
+		ingress:    false,
+	}, log)
+}
+
+func (acl *ACLInterfacesVppCalls) requestSetACLToInterfaces(logicalReq *ACLInterfaceLogicalReq, log logging.Logger) error {
+	var wasErr error
+	for _, aclInterface := range logicalReq.interfaces {
+		// Create acl list with new entry
 		var ACLs []uint32
-		index, _, found := swIfIndexes.LookupIdx(ingressInterface)
+		index, _, found := acl.swIfIndexes.LookupIdx(aclInterface)
 		if !found {
-			log.Debugf("Set interface to ACL: Interface %v not found ", ingressInterface)
+			log.Debugf("Set interface to ACL: Interface %v not found ", aclInterface)
 			continue
 		}
-		// All previously assigned ACLs have to be dumped and added to acl list.
-		aclInterface, err := DumpInterface(index, vppChannel, measure.GetTimeLog(acl_api.ACLInterfaceListDump{}, stopwatch))
+		// All previously assigned ACLs have to be dumped and added to acl list
+		aclInterfaceDetails, err := DumpInterface(index, acl.vppChan, acl.dumpIfaces)
 		if err != nil {
 			return err
 		}
-		// Construct ACL list. ACLs within NInput are defined as ingress, so provided new aclIndex has to be
-		// added to the beginning of the list. todo it would be nicer to add new acl index to newNInput index
-		if aclInterface != nil {
-			ACLs = append(ACLs, aclIndex)
-			for _, aclIndex := range aclInterface.Acls {
-				ACLs = append(ACLs, aclIndex)
+
+		nInput := aclInterfaceDetails.NInput
+		if logicalReq.ingress {
+			// Construct ACL list. ACLs within NInput are defined as ingress, so provided new aclIndex has to be
+			// added to the beginning of the list
+			// TODO it would be nicer to add new acl index to newNInput index
+			if aclInterfaceDetails != nil {
+				ACLs = append(ACLs, logicalReq.aclIndex)
+				for _, aclIndex := range aclInterfaceDetails.Acls {
+					ACLs = append(ACLs, aclIndex)
+				}
+			}
+			nInput++ // Rise NInput
+		} else {
+			// Construct ACL list. ACLs outside of NInput are defined as egress, so provided new aclIndex has to be
+			// added to the end of the list
+			if aclInterfaceDetails != nil {
+				for _, aclIndex := range aclInterfaceDetails.Acls {
+					ACLs = append(ACLs, aclIndex)
+				}
+				ACLs = append(ACLs, logicalReq.aclIndex)
 			}
+			// NInput remains the same
 		}
-		newNInput := aclInterface.NInput + 1 // Rise NInput
 
-		// Measure ACLInterfaceSetACLList time.
+		// Measure ACLInterfaceSetACLList time
 		start := time.Now()
 
-		req := &acl_api.ACLInterfaceSetACLList{}
-		req.Acls = ACLs
-		req.Count = uint8(len(ACLs))
-		req.SwIfIndex = index
-		req.NInput = newNInput
+		msg := &acl_api.ACLInterfaceSetACLList{}
+		msg.Acls = ACLs
+		msg.Count = uint8(len(ACLs))
+		msg.SwIfIndex = index
+		msg.NInput = nInput
 
 		reply := &acl_api.ACLInterfaceSetACLListReply{}
-
-		err = vppChannel.SendRequest(req).ReceiveReply(reply)
+		err = acl.vppChan.SendRequest(msg).ReceiveReply(reply)
 		if err != nil {
-			return fmt.Errorf("failed to set interface %v to ACL %v", ingressInterface, aclIndex)
+			wasErr = err
```

Comment:
Should the loop continue even with error? Even if so, only the last error will be returned to the caller anyway. Also for large number of items that fail the time it will take to finish might be long.